x-pack/plugins/apm/public/components/shared/environment_filter/index.tsx

```diff
@@ -34,76 +38,88 @@ function updateEnvironmentUrl(
-    ...environmentOptions,
-  ];
-}
+// #NOTE do we want ENVIRONMENT_NOT_DEFINED with suggestionSelect
 
 export function ApmEnvironmentFilter() {
-  const { status, environments, environment } = useEnvironmentsContext();
+  const { environment, serviceName, start, end } = useEnvironmentsContext();
 
-  return (
-    <EnvironmentFilter
-      status={status}
+  return serviceName ? (
+    <ServiceEnvironmentFilter
+      start={start}
+      end={end}
       environment={environment}
-      environments={environments}
+      termValue={serviceName}
     />
+  ) : (
+    <EnvironmentFilter start={start} end={end} environment={environment} />
   );
 }
 
 export function EnvironmentFilter({
   environment,
-  environments,
-  status,
+  start,
+  end,
 }: {
   environment: Environment;
-  environments: Environment[];
-  status: FETCH_STATUS;
+  start?: string;
+  end?: string;
 }) {
   const history = useHistory();
   const location = useLocation();
 
-  // Set the min-width so we don't see as much collapsing of the select during
-  // the loading state. 200px is what is looks like if "production" is
-  // the contents.
-  const minWidth = 200;
+  return (
+    <SuggestionsSelect
+      allOption={ENVIRONMENT_ALL}
+      placeholder={i18n.translate('xpack.apm.filter.environment.placeholder', {
+        defaultMessage: 'Select environment',
+      })}
+      prepend={i18n.translate('xpack.apm.filter.environment.label', {
+        defaultMessage: 'Environment',
+      })}
+      onChange={(changeValue) =>
+        updateEnvironmentUrl(history, location, changeValue as string)
+      }
+      defaultValue={getEnvironmentLabel(environment)}
+      field={SERVICE_ENVIRONMENT}
+      start={start}
+      end={end}
+      data-test-subj="environmentFilter"
+    />
+  );
+}
 
-  const options = getOptions(environments);
+export function ServiceEnvironmentFilter({
+  environment,
+  start,
+  end,
+  termValue,
+}: {
+  environment: Environment;
+  start?: string;
+  end?: string;
+  termValue: string;
+}) {
+  const history = useHistory();
+  const location = useLocation();
 
   return (
-    <EuiSelect
-      fullWidth
+    <SuggestionsSelectWithTerm
+      allOption={ENVIRONMENT_ALL}
+      placeholder={i18n.translate('xpack.apm.filter.environment.placeholder', {
+        defaultMessage: 'Select environment',
+      })}
       prepend={i18n.translate('xpack.apm.filter.environment.label', {
         defaultMessage: 'Environment',
       })}
-      options={options}
-      value={environment}
-      onChange={(event) => {
-        updateEnvironmentUrl(history, location, event.target.value);
-      }}
-      isLoading={status === FETCH_STATUS.LOADING}
-      style={{ minWidth }}
+      onChange={(changeValue) =>
+        updateEnvironmentUrl(history, location, changeValue as string)
+      }
+      defaultValue={getEnvironmentLabel(environment)}
+      field={SERVICE_ENVIRONMENT}
+      termField={SERVICE_NAME}
+      termValue={termValue}
```

Comment:
Should this perhaps be:

```suggestion
      fieldName={SERVICE_NAME}
      fieldValue={termValue}
```

Reply:
https://github.com/elastic/kibana/pull/126679/commits/4eae34b4ad02be647d0e51c565a706104c85200c